tidy3d/web/webapi.py

```diff
@@ -99,16 +99,47 @@ def upload(
     To start the simulation running, must call :meth:`start` after uploaded.
     """
 
-    task_id = _upload_task(
-        simulation=simulation,
-        task_name=task_name,
-        folder_name=folder_name,
-        callback_url=callback_url,
-    )
+    simulation.validate_pre_upload()
 
+    json_string = simulation._json_string()  # pylint:disable=protected-access
+    data = {
+        "taskName": task_name,
+        "callbackUrl": callback_url,
+    }
+    folder = _query_or_create_folder(folder_name)
+    method = f"tidy3d/projects/{folder.projectId}/tasks"
+
+    log.debug("Creating task.")
+    try:
+        task = http.post(method=method, data=data)
+        task_id = task["taskId"]
+    except requests.exceptions.HTTPError as e:
+        error_json = json.loads(e.response.text)
+        raise WebError(error_json["error"]) from e
     # log the task_id so users can copy and paste it from STDOUT / file if the need it later.
     log.info(f"Uploaded task '{task_name}' with task_id '{task_id}'.")
 
+    # upload the file to s3
+    log.debug("Uploading the json file")
+    client, bucket, user_id = get_s3_user()
+    key = f"users/{user_id}/{task_id}/simulation.json"
+    log.debug(f"json = {json_string}")
+    resp = client.put_object(
+        Body=json_string,
+        Bucket=bucket,
+        Key=key,
+    )
+    if resp.get("ResponseMetadata") and resp.get("ResponseMetadata").get("HTTPStatusCode") != 200:
+        raise WebError("Upload file to 3S failed, please check your network or try it later.")
+
+    # log the url for the task in the web UI
+    log.info(f"{DEFAULT_CONFIG.website_endpoint}/folders/{folder.projectId}/tasks/{task_id}")
```

Comment:
for me (jupyter notebook), this never displays great: I can't click the link nor can I copy and paste it.  Maybe we can fix it later.

Reply:
True. I'll have a quick look if it can be done better.